flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/KubernetesOperatorConfigOptions.java

```diff
@@ -321,7 +336,32 @@ public static String operatorConfigKey(String key) {
                                     + "but to complement them in special cases. For instance, a "
                                     + "full checkpoint might need to be occasionally triggered to "
                                     + "break the chain of incremental checkpoints and consolidate "
-                                    + "the partial incremental files.");
+                                    + "the partial incremental files. "
+                                    + "WARNING: not intended to be used together with the cron-based "
+                                    + "periodic checkpoint triggering");
+
+    @Documentation.Section(SECTION_DYNAMIC)
+    public static final ConfigOption<String> PERIODIC_CHECKPOINT_CRON =
+            operatorConfig("periodic.checkpoint.cron")
+                    .stringType()
+                    .defaultValue(
+                            "0 0 0 29 2 ? 1999") // Default value that is guaranteed not to trigger (not a leap year)
+                    .withDescription(
+                            "A cron expression that defines when to automatically trigger checkpoints. "
+                                    + "The precise triggering schedule is not guaranteed, checkpoints will "
+                                    + "be triggered as part of the regular reconcile loop. Periodic triggering "
+                                    + "with frequency higher than the reconciliation loop "
+                                    + "("
+                                    + OPERATOR_RECONCILE_INTERVAL.key()
+                                    + ") will lead to trigger omissions. "
+                                    + "NOTE: checkpoints are generally managed by Flink. This "
+                                    + "setting isn't meant to replace Flink's checkpoint settings, "
+                                    + "but to complement them in special cases. For instance, a "
+                                    + "full checkpoint might need to be occasionally triggered to "
+                                    + "break the chain of incremental checkpoints and consolidate "
+                                    + "the partial incremental files. "
+                                    + "WARNING: not intended to be used together with the interval-based "
+                                    + "periodic checkpoint triggering");
```

Comment:
Would the interval and the cron-based logic really not work together, or is this more about warning the user about unwanted checkpoint triggering?

Reply:
For the sake of simplifying this, could we unify both options under `periodic.{checkpoint,savepoint}.interval`? We can parse the String option using `Duration`, if it fails (i.e. throws `DateTimeParseException`) we try the cron-based approach. I think this approach should be safe because a mixup between the two is not possible.

Reply:
I was thinking the same thing, +1 for using the single existing interval config for this.

Reply:
@mxm yes, the warning just relates to the fact that such mixing makes triggering not predictable, because there is just one last checkpoint/savepoint timestamp field that is shared between the two.

Reply:
> For the sake of simplifying this, could we unify both options under periodic.
I am not against it, but see the following downsides:
- We won’t be able to tell if the user misspelled something in the duration format or her cron expression is invalid.  Since Quartz supports both 6- and 7-positional expressions, it can get confusing sometimes when ? or * are used in the "wrong" place. Some combinations are invalid, but we won't able to point at the error with the cron format, since there is a possibility that the user wanted to specify Duration to begin with, but made a mistake there.
- The name would become a bit awkward:  `periodic.{checkpoint|savepoint}.interval` , whereas cron does not necessarily represent an interval, but an also be a point-in-time setting.  This probably leads to the necessity to deprecate the already-released `periodic.savepoint.interval`
- Although the absolutely valid intention is to make things simple for the user, the complexity of explaining the "Eierlegende Milchwollsau" functionality of the single setting in the docs plus the unclear naming seem to work against this intention.

Reply:
After thinking about this a bit more, I guess a generic error message that "something is wrong with the format, please make sure to either use a valid Duration data type of a cron expression" should also be OK.

Reply:
@mxm, @gyfora 
PTAL https://github.com/apache/flink-kubernetes-operator/commit/412f9136c5a26b3160b50aefb7e19e4430d9c0f8